warehouse/admin/views/users.py

```diff
@@ -301,6 +307,35 @@ def user_add_email(user, request):
     return HTTPSeeOther(request.route_path("admin.user.detail", username=user.username))
 
 
+@view_config(
+    route_name="admin.user.delete_email",
+    require_methods=["POST"],
+    permission=Permissions.AdminUsersEmailWrite,
+    uses_session=True,
+    require_csrf=True,
+    context=User,
+)
+def user_email_delete(user: User, request: Request) -> HTTPSeeOther:
+    email = request.db.scalar(
+        select(Email).where(Email.email == request.POST.get("email_address"))
```

Comment:
This would remove the need to check against `user.emails`:
```suggestion
        select(Email).where(_and(
            Email.email == request.POST.get("email_address"),
            Email.user_id == user.id,
        )
```

Reply:
Great idea, simplifies a bit. Commit incoming!